homeassistant/components/iaqualink/switch.py

```diff
@@ -0,0 +1,66 @@
+"""Support for Aqualink pool feature switches."""
+import logging
+
+from iaqualink import AqualinkToggle
+
+from homeassistant.components.switch import DOMAIN, SwitchDevice
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
+
+from . import AqualinkEntity, refresh_system
+from .const import DOMAIN as AQUALINK_DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, config_entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up discovered switches."""
+    devs = []
+    for dev in hass.data[AQUALINK_DOMAIN][DOMAIN]:
+        devs.append(HassAqualinkSwitch(dev))
+    async_add_entities(devs, True)
+
+
+class HassAqualinkSwitch(SwitchDevice, AqualinkEntity):
+    """Representation of a switch."""
+
+    def __init__(self, dev: AqualinkToggle):
+        """Initialize the switch."""
+        SwitchDevice.__init__(self)
+        self.dev = dev
+
+    @property
+    def name(self) -> str:
+        """Return the name of the switch."""
+        return self.dev.label
+
+    @property
+    def icon(self) -> str:
+        """Return an icon based on the switch type."""
+        if self.name == "Cleaner":
```

Comment:
What kind of switches are these? We have specific entity components for vacuum and fan.

Reply:
They're essentially on/off toggles.

Reply:
Ok.